src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -214,129 +429,190 @@ performFetchWithCompletionHandler:(void (^)(UIBackgroundFetchResult result))comp
 
-```bash
-$ cd some/path/my_flutter
-$ flutter attach
-Waiting for a connection from Flutter on iPhone X...
+<?code-excerpt "Creating engine" title?>
+```objectivec
+FlutterEngine *flutterEngine = [[FlutterEngine alloc] initWithName:@"my flutter engine"];
+[[flutterEngine navigationChannel] invokeMethod:@"setInitialRoute" arguments:@"/onboarding"];
+[flutterEngine run];
+@end
 ```
 
-Launch `MyApp` in debug mode from Xcode. Navigate to an area of the app that uses
-Flutter. Then turn back to the terminal, and you should see output similar to the
-following:
+</div>
 
-```text
-Done.
-Syncing files to device iPhone X...                          4.7s
+<div class="tab-pane" id="initial-route-swift-tab" role="tabpanel" aria-labelledby="initial-route-swift-tab" markdown="1">
 
-🔥  To hot reload changes while running, press "r". To hot restart (and rebuild state), press "R".
-An Observatory debugger and profiler on iPhone X is available at: http://127.0.0.1:54741/
-For a more detailed help message, press "h". To quit, press "q".
+<?code-excerpt "Creating engine" title?>
+```swift
+let flutterEngine = FlutterEngine(name: "my flutter engine")
+flutterEngine.navigationChannel.invokeMethod("setInitialRoute", arguments:"/onboarding")
+flutterEngine.run()
 ```
 
-You can now edit the Dart code in `my_flutter`, and the changes can be hot
-reloaded by pressing `r` in the terminal. You can also paste the URL above into
-your browser to use the Dart Observatory for setting breakpoints, analyzing
-memory retention and other debugging tasks.
+</div>
+
+</div>{% comment %} Initial route language tab end {% endcomment -%}
+
+will set your `dart:ui`'s [`window.defaultRouteName`]({{site.api}}/flutter/dart-ui/Window/defaultRouteName.html)
+to `"/onboarding"` instead of `"/"`.
+
+{{site.alert.note}}
+`"setInitialRoute"` on the `navigationChannel` must be called before running your
+`FlutterEngine` in order for Flutter's very first frame to use the desired
+route.
+{{site.alert.end}}
+
+{{site.alert.tip}}
+In order to change your Flutter route after the `FlutterEngine` is already
+running, use [pushRoute]({{site.api}}/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)pushRoute:)
+or [popRoute]({{site.api}}/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)popRoute)
+on the `FlutterViewController`.
+
+To pop the iOS route from the Flutter side, call [`SystemNavigator.pop()`]({{site.api}}/flutter/services/SystemNavigator/pop.html)
+{{site.alert.end}}
+
+See the [Navigation and routing](/docs/development/ui/navigation) section for more on Flutter's routes.
+
+### Other
+
+The above only illustrates a few examples of ways to customize how a Flutter
+instance is initiated. Using [platform channels](/docs/development/platform-integration/platform-channels),
+you're free to push data or prepare your Flutter environment in any way you'd
+like before presenting the Flutter UI via a `FlutterViewController`.
```

Comment:
IME, the routing logic for add-to-app is pretty limited and does unexpected things. I think it would be better to document how to use platform channels in this way here, and then we could mention that you can use any of the framework-provided channels as well (such as the navigation channel).

In other words, I think a user is far more likely to want to tell the running Dart program what state it should initialize to before presenting, and that it's very unlikely that the right way to do that will be setting an initial route (that only works at a specific point in execution) or pushing/popping routes blindly from the background.

Reply:
Document platform channels in this way here -> document how to use a custom platform channel to communicate with Dart

Reply:
I generally agree with you and using a custom channel to push an application specific DI-like bundle is what I would do too. 

Though on the flip side, there's a bit of a personal taste involved in the trade off in the sense that writing platform channels is extremely boring and tedious (and error-prone in its own right as well) currently until we do https://github.com/flutter/flutter/issues/32930. 

Routes is also a non-trivially important concept in Flutter and we're not necessarily going to be avoiding pains for developers if we avoided this whole topic in documentations. 

TL;DR, I think all present solutions have annoyances and we should properly handle this very common initialization use case and present a proper best practice with infrastructure in the coming phases, but we should have some docs in the meantime.

Reply:
Added a bit more caveating for more transparency in the docs in the meantime.

Reply:
oh look there's https://github.com/flutter/flutter/issues/34170 already

Reply:
SG.  I can understand including this - my comment above sketches out a bit more of why (I think users will read this, get it wrong, think it's a bug, and be frustrated when we tell them it's by design).  But maybe that's what we have right now.

Reply:
ya, I hear you. There are tons of bugs around this already and I don't think routes are generally useful APIs, but I don't think we can move the weight of the community right now and we might get more confusion if the code is in the repo but we avoided talking about it.